Add delete events feature

macdeath-AA opened the pull request.

macdeath-AA committed:
Add delete events feature

tarun-menta commented on src/App.js:
```diff
@@ -84,6 +84,30 @@ function App() {
+    const jsonObj = JSON.parse(localStorage.getItem(customKey)) || [];
+
+    const newCustomList = [];
+    jsonObj.forEach((delEvent) => {
+      console.log(startTime, delEvent.title);
+
+      if (
+        delEvent.title === eventName
+        && delEvent.start === startTime
+        && delEvent.end === endTime
+      ) {
+        return;
+      }
+      newCustomList.push(delEvent);
+    });
```
This _could_ be replaced by a filter method, but it probably won't impact performance anyways

tarun-menta reviewed (changes requested):
Course events are not getting deleted. This is because course events are stored in _both_ `masterkey` and `aimskey` in localStorage. Simple fix is to iterate through the JSON stored in `aimskey` too, and repeat the same deletion code there too

butter-chicken27 commented on src/App.js:
```diff
@@ -84,6 +84,30 @@ function App() {
+        return;
+      }
+      newCustomList.push(delEvent);
+    });
```
Get rid of the unnecessary console logs. Variables like jsonObj need to be changed to more meaningful names.

RachitKeertiDas replied:
I agree. Variable names like `delEvent` can be improved. Especially because `delEvent` is actually the event which stays and is *not* being deleted!

A generic name for the function argument like `event` might suffice.

butter-chicken27 commented on src/pages/TimeTable.js:
```diff
@@ -30,6 +31,7 @@ function TimeTable({ eventList, handleNewCustomEvent }) {
   let eventDate = null;
   let startTime = null;
   let endTime = null;
+  // let note = "";
```
Delete the comment

butter-chicken27 commented on src/pages/TimeTable.js:
```diff
@@ -214,6 +235,13 @@ function TimeTable({ eventList, handleNewCustomEvent }) {
+          label="Add note"
+          fullWidth
+          onChange={handleNoteChange}
+          /> */}
```
_Another_ comment to be deleted

macdeath-AA commented:
oooo yea i forgot to beautify this my bad. Will do.

On Mon, Apr 11, 2022 at 1:14 PM butter-chicken27 ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
> ------------------------------
>
> In src/App.js
> <https://github.com/LambdaIITH/iith-dashboard-pwa/pull/18#discussion_r847031102>
> :
>
> > +  const removeEvent = (eventName, startTime, endTime) => {
> +    const jsonObj = JSON.parse(localStorage.getItem(customKey)) || [];
> +
> +    const newCustomList = [];
> +    jsonObj.forEach((delEvent) => {
> +      console.log(startTime, delEvent.title);
> +
> +      if (
> +        delEvent.title === eventName
> +        && delEvent.start === startTime
> +        && delEvent.end === endTime
> +      ) {
> +        return;
> +      }
> +      newCustomList.push(delEvent);
> +    });
>
> Get rid of the unnecessary console logs. Variables like jsonObj need to be
> changed to more meaningful names.
> ------------------------------
>
> In src/pages/TimeTable.js
> <https://github.com/LambdaIITH/iith-dashboard-pwa/pull/18#discussion_r847032528>
> :
>
> > @@ -30,6 +31,7 @@ function TimeTable({ eventList, handleNewCustomEvent }) {
>    let eventDate = null;
>    let startTime = null;
>    let endTime = null;
> +  // let note = "";
>
> Delete the comment
> ------------------------------
>
> In src/pages/TimeTable.js
> <https://github.com/LambdaIITH/iith-dashboard-pwa/pull/18#discussion_r847033597>
> :
>
> > @@ -214,6 +235,13 @@ function TimeTable({ eventList, handleNewCustomEvent }) {
>                shrink: true,
>              }}
>            />
> +          {/* <TextField
> +          margin="dense"
> +          id="notes"
> +          label="Add note"
> +          fullWidth
> +          onChange={handleNoteChange}
> +          /> */}
>
> *Another* comment to be deleted
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/LambdaIITH/iith-dashboard-pwa/pull/18#pullrequestreview-937584866>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AUL67E7ZUGAAVNXUGH2HVZLVEPJ6DANCNFSM5S6VEHBQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

-- 


Disclaimer:- This footer text is to convey that this email is sent by one 
of the users of IITH. So, do not mark it as SPAM.

macdeath-AA committed:
Display error message on deleting AIMs event
- Windows error popup when trying to delete AIMS event
- Changed variable names
- Removed console logs and unnecessary comments

RachitKeertiDas commented on src/pages/TimeTable.js:
```diff
@@ -83,18 +83,20 @@ function TimeTable({ eventList, handleNewCustomEvent, handleDeleteEvent }) {
-      clickInfo.event.remove();
-      handleDeleteEvent(
+
+    const aimsTitle = aimsCourse.identifiedCourses.find(course => course === clickInfo.event.title)
```
This is not going to work in the newer version of the extension, where the course title also includes the course name.

RachitKeertiDas commented on src/pages/TimeTable.js:
```diff
@@ -83,18 +83,20 @@ function TimeTable({ eventList, handleNewCustomEvent, handleDeleteEvent }) {
+    if (aimsTitle !== undefined){
+      window.alert('AIMS courses cannot be deleted.');
+    }
+    else {
```
Indentation Error

RachitKeertiDas reviewed (changes requested):
It seems that the eslint and prettier hooks were not run, again.
There is also an eslint issue (react/forbid-prop-types) causing some issues atm.

Also, the AIMS Error mesages sdoes not show up if you're using the newer extension, in which the title also contains the course names.

macdeath-AA committed:
Added recurring events for AIMS courses
